GERTi/GERTiClient.lua

```diff
@@ -201,57 +292,71 @@ handler["RemoveNeighbor"] = function (eventName, receivingModem, sendingModem, p
 end
 
 handler["RegisterNode"] = function (eventName, receivingModem, sendingModem, port, distance, code, origination, tier, serialTable)
+	-- THIS IS COMPLETELY IGNORED. Until I have more knowledge about the organization, I won't touch this,
+	--  but FIX THIS LATER! Also see RegisterComplete and the stuff that happened to ResolveAddress - 20kdc
 	transmitInformation(neighbors[1]["address"], neighbors[1]["port"], "RetrieveAddress", origination, tier, serialTable)
 	local _, _, _, port, _, payload = event.pull(5, "modem_message")
 	if payload ~= nil then
 		return transmitInformation(sendingModem, port, payload)
 	end
 end
 
+handler["RegisterComplete"] = function (eventName, receivingModem, sendingModem, port, distance, code, targetMA, iResponse)
+	if receivingModem == targetMA then
+		-- This is us!
+		iAddress = iResponse
+	end
+	-- The code that goes in the 'else' depends on what happens to RegisterNode
+end
+
 handler["RETURNSTART"] = function (eventName, receivingModem, sendingModem, port, distance, code, tier)
 	-- Store neighbor based on the returning tier
-	return storeNeighbors(eventName, receivingModem, sendingModem, port, distance, tier)
+	storeNeighbors(eventName, receivingModem, sendingModem, port, distance, tier)
+end
+
+handler["ResolveAddress"] = function (eventName, receivingModem, sendingModem, port, distance, code, gAddress)
+	addWaitingAddress(gAddress, function(response)
+		-- this indentation is weird.
+		-- Anyway, this is the callback to forward the resolution
+		transmitInformation(sendingModem, port, "AddressResolution", gAddress, response)
+	end)
+end
+
+handler["AddressResolution"] = function (eventName, receivingModem, sendingModem, port, distance, code, gAddress, mtAddress)
+	if waitingAddress[gAddress] then
+		waitingAddress[gAddress](mtAddress)
+		waitingAddress[gAddress] = nil
+	end
 end
 
 local function receivePacket(eventName, receivingModem, sendingModem, port, distance, code, ...)
 	print(code)
 	-- Attempt to call a handler function to further process the packet
 	if handler[code] ~= nil then
-		return handler[code](eventName, receivingModem, sendingModem, port, distance, code, ...)
+		handler[code](eventName, receivingModem, sendingModem, port, distance, code, ...)
 	end
+	-- Just to make it clear, the return value system was a bad idea.
+end
+
+local function addressResolver(gAddress)
```

Comment:
Please fold in the cached check to this, as GERTi.resolveAddress no longer exists.